### PR TITLE
Task 22 Add EF for Diary microservice

### PR DESCRIPTION
**Pull request by Ainur555**
I added EF for Diary, added basic projects for further work and fixed the last PR comments

### CODE REVIEW

File: Diary/Domain.Entities/Domain/BaseTypes/RelatedEntityTypes.cs [hunk 1]
```diff
@@ -4,11 +4,11 @@
 using System.Text;
 using System.Threading.Tasks;
 
-namespace Domain.Entities.BaseTypes
+namespace Diary.Core.Domain.BaseTypes
 {
     public enum RelatedEntityTypes
```
Comment (YauhenIlyushenka) [inline Diary/Domain.Entities/Domain/BaseTypes/RelatedEntityTypes.cs]: Enum must have a singular type name - https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#naming-enumerations

Comment (Ainur555) [reply Diary/Domain.Entities/Domain/BaseTypes/RelatedEntityTypes.cs]: thanks, i am gonna fix

Comment (Ainur555) [reply Diary/Domain.Entities/Domain/BaseTypes/RelatedEntityTypes.cs]: its done

**Review by YauhenIlyushenka — APPROVED** [review]

File: Diary/Diary.DataAccess/Abstractions/BaseRepository.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        }
+
+        /// <summary>
+        /// Запросить все сущности в базе.
+        /// </summary>
+        /// <param name="asNoTracking"> Вызвать с AsNoTracking. </param>
+        /// <returns> IQueryable массив сущностей. </returns>
+        public virtual IQueryable<T> GetAll(bool asNoTracking = false)
+        {
+            return asNoTracking ? _entitySet.AsNoTracking() : _entitySet;
```
Comment (AATerekhov) [inline Diary/Diary.DataAccess/Abstractions/BaseRepository.cs:41]: How safe is it to transfer non-materialized entities from the repository? The principle of sole responsibility is violated here. Ideally, return only IEnumerable.

File: Diary/Diary.DataAccess/Abstractions/BaseRepository.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        }
+
+
+        /// <summary>
+        /// Для сущности проставить состояние - что она изменена.
+        /// </summary>
+        /// <param name="entity"> Сущность для изменения. </param>
+        public void Update(T entity)
+        {
+            Context.Entry(entity).State = EntityState.Modified;
```
Comment (AATerekhov) [inline Diary/Diary.DataAccess/Abstractions/BaseRepository.cs:88]: I'm not sure that using non-asynchronous code is a good practice.

**Review by AATerekhov — COMMENTED** [review]
There are several recommendations

File: Diary/Diary.DataAccess/Abstractions/BaseRepository.cs (hunk 3 above)
Comment (YauhenIlyushenka) [reply Diary/Diary.DataAccess/Abstractions/BaseRepository.cs:88]: @AATerekhov it is a simple operation and we do not have any benefits from using async method in that case.

**Review by AATerekhov — APPROVED** [review]
My recommendations, just for information.

**Review by VladimirFKulagin — APPROVED** [review]

**Pull request merged**